src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import java.util.ArrayList;
+
+import task.Task;
+
+/**
+ * Encapsulates a taskList. An array full of Tasks object.
+ *
+ *  @author Tan Qin Yong
+ */
+public class TaskList {
+    /** The list containing tasks. */
+    private ArrayList<Task> taskList;
+
+    /** The count of tasks in the list. */
+    private int count;
+
+    /**
+     * Constructor of taskList.
+     *
+     * @param taskList A given list of tasks to initialize the duke.TaskList.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+        count = taskList.size();
+    }
+
+    /**
+     * Constructs an empty duke.TaskList.
+     */
+    public TaskList() {
+        taskList = new ArrayList<Task>();
+        count = 0;
+    }
+
+    /**
+     * Adds a new task to the end of the list.
+     *
+     * @param newTask The new task provided by the user.
+     * @param loadTask Indicates whether task is being loaded by IO.
+     */
+    public void addTask(Task newTask, boolean loadTask) {
+        taskList.add(newTask);
+        count++;
+
+        if (!loadTask) {
+            System.out.println("Cool! Adding new task: ");
+            System.out.println(newTask.toString());
+            System.out.println("Now you have " + this.count
+                    + " tasks in your list.");
```

Comment:
Consider breaking the print statements into a separate method for better separation of concerns, as you did for your duke task with ui.printLine()

```suggestion
            UI.printLoadTask;
```